### PR TITLE
Issue_2195_disabled undo button when starting organelle moving

### LINKED ISSUE #2195: Undo and redo should be unavailable while moving a part
otherwise the undo can be exploited to get infinite MP.

https://community.revolutionarygamesstudio.com/t/can-cheat-the-editor-for-unlimited-points/3473

### PR DESCRIPTION
**Pull request by Snacklemore**
Hello,
Pretty easy fix, just disabled the undo Button when moving started. I thought I would have to  nable it afterwards when placing,  
but the update when placing seems to do it already.

Closes #2195

### CODE REVIEW

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 1]
```diff
@@ -781,6 +781,9 @@ public void StartOrganelleMove(OrganelleTemplate selectedOrganelle)
 
         MovingOrganelle = selectedOrganelle;
         editedMicrobeOrganelles.Remove(MovingOrganelle);
+
+        // disable undo button while moving(enable when placing)
+        gui.SetUndoButtonStatus(false);
```
Comment (Kasterisk) [inline src/microbe_stage/editor/MicrobeEditor.cs:786]: The redo button need to be disabled here also.

**Review by Kasterisk — CHANGES_REQUESTED** [review]
The undo/redo could still be accessed via shortcut keys. I think it's good if they're also disabled by adding guards in `Undo()` and `Redo()` method in the editor class that checks whether `MovingOrganelle` is null, and returns if it's not.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
Doesn't work correctly with saving:
- Start an organelle move
- Save and then immediately load
- Notice that the undo button is enabled

**Review by Kasterisk — APPROVED** [review]
Improved the functionality and made this work with saving.

**Comment by hhyyrylainen** [comment]
Looking at your changes @Kasterisk it almost looks like you rewrote the entire PR... If you did, could you make a separate PR just from your commit?

**Pull request closed without merging**